### PR TITLE
* : remove error rate

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/46056

Problem Summary:

### What is changed and how it works?

Remove the error rate struct, it was for the feedback feature which already has been deprecated.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: statistics/index.go [hunk 1]
```diff
@@ -115,23 +114,20 @@ func (idx *Index) IsInvalid(sctx sessionctx.Context, collPseudo bool) (res bool)
...
-				"NotAccurate", notAccurate,
 				"TotalCount", totalCount,
 			)
 			debugtrace.LeaveContextCommon(sctx)
 		}()
 	}
-	notAccurate = idx.ErrorRate.NotAccurate()
 	totalCount = idx.TotalRowCount()
-	return (collPseudo && notAccurate) || totalCount == 0
+	return (collPseudo) || totalCount == 0
```
Comment (hi-rustin) [inline statistics/index.go:130]: I am not sure whether this makes sense.

**Review by qw4990 — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

File: statistics/column.go [hunk 2]
```diff
@@ -133,7 +136,7 @@ func (c *Column) IsInvalid(sctx sessionctx.Context, collPseudo bool) (res bool)
 			debugtrace.LeaveContextCommon(sctx)
 		}()
 	}
-	if collPseudo && c.NotAccurate() {
+	if collPseudo {
```
[the remark this answers is not in the document]
Comment (time-and-fate) [reply statistics/column.go:139]: Yes.
Without feedback, `NotAccurate()` should always return `true`. This change shouldn't cause any actual difference.

File: statistics/index.go (hunk 1 above)
Comment (time-and-fate) [reply statistics/index.go:130]: ditto.

**Pull request merged**